res/qml/Deck.qml

```diff
@@ -35,380 +161,708 @@ Item {
+                }
+            }
+        }
+        DelegateChoice {
+            roleValue: "hotcueAndStem"
+            LayoutItem {
+                Layout.fillWidth: true
+                Layout.minimumWidth: 250
+                height: 92
+
+                blurRadius: 4
+                editOverlay.color: Qt.alpha('black', 0.5)
+                editLabel.text: "Hotcue & Stem"
+                editLabel.font.pixelSize: 14
+                editLabel.color: Theme.white
+
+                DeckComponent.HotcueAndStem {
                     group: root.group
-                    width: playButton.height
-                    height: playButton.height
+                    currentTrack: root.currentTrack
+                    anchors.fill: parent
                 }
             }
         }
+    }
 
-        Skin.SyncButton {
-            id: syncButton
+    ColumnLayout {
+        id: grid
 
-            anchors.right: parent.right
-            anchors.top: parent.top
-            group: root.group
+        anchors.fill: parent
+        anchors.margins: 4
+
+        visible: !root.minimized
+
+        Repeater {
+            model: itemModel
+            delegate: deckItemDelegate
         }
 
         FadeBehavior on visible {
-            fadeTarget: buttonBar
+            fadeTarget: grid
         }
     }
 
-    Mixxx.PlayerDropArea {
+    ColumnLayout {
+        id: minimizedGrid
+
         anchors.fill: parent
+        anchors.margins: 4
+
+        visible: root.minimized
+
+        Repeater {
+            model: minimizedItemModel
+            delegate: deckItemDelegate
+        }
+
+        FadeBehavior on visible {
+            fadeTarget: minimizedGrid
+        }
+    }
+
+    Mixxx.PlayerDropArea {
         group: root.group
+        anchors.fill: parent
+
+        property var candidate: null
+
+        onEntered: {
+            candidate = null
+        }
+
+        onPositionChanged: (drag) => {
+            let parentArea = drag.source.parent.mapToItem(grid, 0, 0, drag.source.parent.width, drag.source.parent.height)
+            let position = Qt.point(drag.x, drag.y)
+            if (position.x < parentArea.x || position.x >= parentArea.x + parentArea.width || position.y < parentArea.y || position.y > parentArea.y + parentArea.height) {
+                candidate = null
+                return
+            }
+
+            if (!drag.source.parent.items) {
+                console.error(`No items list on the parent of ${drag.source}`)
+                candidate = null
+                return
+            }
+            let target = null
+            let sourceIdx
+            for (let i = 0; i < drag.source.parent.children.length; i++) {
+                if (drag.source.parent.children[i] == drag.source) {
+                    sourceIdx = i;
+                    break
+                }
+            }
+            if (sourceIdx === null) {
+                console.error("Cannot find the source item in the parent's children")
```

Comment:
I see a mix of semicolons and no semicolons.  We should use a linter to enforce style

Reply:
ditto qmllint :(